modules/system/console/WinterTest.php

```diff
@@ -0,0 +1,112 @@
+<?php namespace System\Console;
+
+use Illuminate\Console\Command;
+use Symfony\Component\Process\Exception\ProcessSignaledException;
+use Symfony\Component\Process\Process;
+use System\Classes\PluginManager;
+use Symfony\Component\Console\Input\InputArgument;
+
+/**
+ * Console command to test a plugin or winter's core.
+ *
+ * This searches for a phpunit.xml file into the plugin's directory and run its tests
+ *
+ * @package winter\wn-system-module
+ */
+class WinterTest extends Command
+{
+    /**
+     * The console command name.
+     * @var string
+     */
+    protected $name = 'winter:test';
+
+    protected $signature = 'winter:test {plugin?}';
+
+    /**
+     * The console command description.
+     * @var string
+     */
+    protected $description = "Run tests of Winter's core or an existing plugin.";
+
+    protected $exec = ['./vendor/bin/phpunit'];
+
+    /**
+     * Create a new command instance.
+     *
+     * @return void
+     */
+    public function __construct()
+    {
+        parent::__construct();
+
+        /**
+         * Ignore validation errors as option proxying is used by this command
+         * @see https://github.com/nunomaduro/collision/blob/stable/src/Adapters/Laravel/Commands/TestCommand.php
+         */
+        $this->ignoreValidationErrors();
+    }
+
+    /**
+     * Execute the console command.
+     * @return int
+     */
+    public function handle()
+    {
+        // If the plugin argument is set, search for it and run its tests
+        if ($pluginArgument = $this->argument('plugin')) {
+            $pluginManager = PluginManager::instance();
+            $pluginName = $pluginManager->normalizeIdentifier($pluginArgument);
+
+            if (!$pluginManager->exists($pluginName)) {
+                throw new \InvalidArgumentException(sprintf('Plugin "%s" not found.', $pluginName));
```

Comment:
@RomainMazB would it be better to just simply output an error as opposed to throwing an exception? A missing plugin is a fixable error.
```suggestion
                $this->error(sprintf('Plugin "%s" not found.', $pluginName));
```